src/registry/children/ChildModal.tsx

```diff
@@ -0,0 +1,78 @@
+import {PrimaryActionButton} from '@utrecht/component-library-react';
+import {Formik, useFormikContext} from 'formik';
+import {FormattedMessage, useIntl} from 'react-intl';
+import {toFormikValidationSchema} from 'zod-formik-adapter';
+
+import FormFieldContainer from '@/components/FormFieldContainer';
+import Modal from '@/components/modal';
+
+import {BSNField, DateOfBirthField, FirstNamesField} from './subFields';
+import type {ExtendedChildDetails} from './types';
+import {buildChildSchema} from './validationSchema';
+
+const ChildModalSaveButton = () => {
+  const {submitForm} = useFormikContext<unknown>();
+  return (
+    <PrimaryActionButton
+      type="button"
```

Comment:
with the reworked modal (rendering in a portal), you can use `type="submit"` and substantially simplify things -  `useFormikContext` is no longer necessary then